packages/jest-snapshot/src/utils.ts

```diff
@@ -185,8 +185,6 @@ const deepMergeArray = (target: Array<any>, source: Array<any>) => {
   source.forEach((element, index) => {
     if (typeof mergedOutput[index] === 'undefined') {
       mergedOutput[index] = element;
-    } else if (typeof element === 'undefined') {
```

Comment:
In adding tests to cover this line, I realized that this case really shouldn't happen (i.e. `source` having an array element that's undefined)